node/src/components/linear_chain.rs

```diff
@@ -407,9 +424,28 @@ where
                     era_id,
                     ..
                 } = *fs;
-                if let Err(err) = fs.verify() {
-                    warn!(%block_hash, %public_key, %err, "received invalid finality signature");
-                    return Effects::new();
+                if let Some(latest_block) = self.latest_block.as_ref() {
+                    // If it's a switch block it has already forgotten its own era's validators,
+                    // unbonded some old validators, and determined new ones. In that case, we
+                    // should add 1 to last_block_era.
+                    let current_era = latest_block.header().era_id()
+                        + if latest_block.header().is_switch_block() {
+                            1
+                        } else {
+                            0
+                        };
+                    let lowest_acceptable_era_id =
+                        current_era + self.auction_delay - self.unbonding_delay;
```

Comment:
`subtract with overflow`
Makes sense. :smile: 
```suggestion
                        (current_era + self.auction_delay).saturating_sub(self.unbonding_delay);
```

Reply:
Thanks :) 

I ran `cargo clippy` locally but it didn't return those warnings.

Reply:
They're not enabled yet: The overflow actually _happened_, in the tests.